src/library_gl.js

```diff
@@ -1871,17 +1871,19 @@ var LibraryGL = {
         break;
     }
 #endif
-    if (!data) {
-      GLctx.bufferData(target, size, usage);
-    } else {
 #if USE_WEBGL2
-      if (GL.currentContext.supportsWebGL2EntryPoints) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
+    if (GL.currentContext.supportsWebGL2EntryPoints) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
+      if (data) {
         GLctx.bufferData(target, HEAPU8, usage, data, size);
-        return;
+      } else {
+        GLctx.bufferData(target, size, usage);
       }
+    } else {
 #endif
-      GLctx.bufferData(target, HEAPU8.subarray(data, data+size), usage);
+      GLctx.bufferData(target, data ? HEAPU8.subarray(data, data+size) : size, usage);
```

Comment:
the collapsing here makes the site polymorphic (VM will see an array sometimes, a size other times). that might cause deopts. is one of the paths unlikely? if so sounds good, and worth adding a comment on it.